### PR TITLE
Game Rating Feature

### PR DESCRIPTION
**Pull request by ryvrook**
Semi complete rating feature.  Needs another Ajax call for an update, and the post needs to be adjusted to actually add value to user so it saves. (Do Not Merge until we work through kinks)

Resolves issue #44

### CODE REVIEW

**Comment by benwagner813** [comment]
I made some changes, the star ratings create/update database records now. Stars don't show for users who aren't logged in. 

When you go back to the page after already having rated the game your rating is "remembered" but hovering over it deselects all the stars (minor problem). 

We still need something that will calculate the average rating for a game and we should add a corresponding column to the game table.

**Comment by oldskoolfan** [comment]
@Voxel20 @benwagner813 @beemteam2 @mihaelavalac make sure to have a look at [my commit](https://github.com/csci-2479-sp-2022/tabletop-nerds/pull/60/commits/50d9ee70ec5c1cd8aabe7ef2b878db9975ccde39), you need to have the `RefreshDatabase` trait on your test classes if you're going to interact with the database, and you should run the seeder in `setUp()` if you're expecting test data to exist. Let me know if you have any questions.

File: app/Contracts/AccountInterface.php [hunk 1]
```diff
@@ -6,5 +6,6 @@ interface AccountInterface
 {
     function getUserWishlist(int $user_id);
     function getWishlistGameById(int $id);
-
+    function getUserRating(?int $user_id, int $game_id);
```
Comment (mihaelavalac) [inline app/Contracts/AccountInterface.php:9]: Added the functions from service to interface.

File: app/Http/Controllers/GameController.php [hunk 2]
```diff
@@ -28,10 +30,11 @@ public function showGame(Request $request, $id)
         $user = $request->user();
         $user ? $user = $user->id : $user = null; //login check
         $reviewed = Review::where('user_id', $user)->where('game_id', $id)->first() ? true : false;
-
+        $userRating = $this->accountInterface->getUserRating($user, $id);
+        $userRating ? $userRating = $userRating->game_rating : null;
         $game ?? throw new NotFoundHttpException();
 
-        return view('game-info', [ 'game' => $game, 'reviews' => $reviews, 'reviewed' => $reviewed]);
+        return view('game-info', [ 'game' => $game, 'reviews' => $reviews, 'reviewed' => $reviewed, 'userRating' => $userRating]);
```
Comment (mihaelavalac) [inline app/Http/Controllers/GameController.php:37]: Added rating to the view.

File: app/Models/Ratings.php [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+    {
+        return $this->belongsTo(Game::class);
+    }
+
+    public function user()
+    {
+        return $this->belongsTo(User::class);
+    }
+
+    public static function initRating($game_id){
```
Comment (mihaelavalac) [inline app/Models/Ratings.php:24]: Return rating for the authenticated user.

File: app/Models/User.php [hunk 4]
```diff
@@ -50,4 +50,8 @@ public function reviews()
     public function userWishlist(){
         return $this->hasMany(Wishlist::class);
     }
+
+    public function userRating(){
```
Comment (mihaelavalac) [inline app/Models/User.php:54]: Added relationship between User and Rating.

File: database/migrations/2022_04_10_164523_create_ratings_table.php [hunk 5]
```diff
@@ -13,12 +13,12 @@
      */
     public function up()
     {
-        Schema::create('game_user', function (Blueprint $table) {
+        Schema::create('ratings', function (Blueprint $table) {
```
Comment (mihaelavalac) [inline database/migrations/2022_04_10_164523_create_ratings_table.php:16]: Rating table.

File: tests/Feature/AccountControllerTest.php [hunk 6]
```diff
@@ -9,9 +9,12 @@
 use App\Models\Game;
 use App\Models\Review;
 use App\Models\User;
+use Illuminate\Foundation\Testing\RefreshDatabase;
```
Comment (mihaelavalac) [inline tests/Feature/AccountControllerTest.php:12]: Thanks, @oldskoolfan, for updating the test.

**Review by mihaelavalac — APPROVED** [review]

**Review by mihaelavalac — APPROVED** [review]

**Pull request merged**